### PR TITLE
Implement shrine template

### PR DESCRIPTION
**Pull request by blasten**
Shrine Template is here!  cc @keanulee

### CODE REVIEW

File: app-header/app-header.html [hunk 1]
```diff
@@ -298,18 +335,20 @@
           var scrollTop = this._clampedScrollTop;
           var savedDisabled = this.disabled;
           this._height = this.offsetHeight;
-          this._topbar = Polymer.dom(this).querySelector('app-toolbar');
+          this._topbar = this._getTopBar();
```
Comment (keanulee) [inline app-header/app-header.html]: `_topbar` or `_topBar`? Should be consistent, here and elsewhere

File: templates/shrine/src/shrine-app.html [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    <iron-ajax url="../data/items.json" auto last-response="{{items}}"></iron-ajax>
+    <iron-ajax url="../data/featured.json" auto last-response="{{featuredItems}}"></iron-ajax>
+    <iron-page-url hash="{{urlHash}}"></iron-page-url>
+    <iron-media-query query="max-width: 400px" query-matches="{{smallScreen}}"></iron-media-query>
+
+    <app-drawer-layout drawer-width="288px" force-narrow>
+      <!-- navigation drawer for small screen sizes -->
+      <app-drawer id="drawer" swipe-open>
+        <app-toolbar class="profileBar">
+          <img class="profilePic" src="//www.gstatic.com/angular/material-adaptive/shrine/16c477b.jpg"  width="30" height="30">
```
Comment (keanulee) [inline templates/shrine/src/shrine-app.html]: Not app-layout-assets?

File: app-header/app-header.html [hunk 3]
```diff
@@ -409,8 +447,8 @@
        */
       _transformHeader: function(y) {
         this.transform('translate3d(0,' + (-y) + 'px, 0)');
```
Comment (keanulee) [inline app-header/app-header.html]: Nit: should this use `this.translate3d()` as well?

File: app-header/app-header.html [hunk 4]
```diff
@@ -284,6 +301,25 @@
...
+        Polymer.dom(this.$.content).getDistributedNodes().forEach(function(node) {
+          if (node.nodeType === Node.ELEMENT_NODE) {
+            if (node.hasAttribute('primary')) {
+              topBar = node;
+            } else if (!topBar) {
+              topBar = node;
+            }
+          }
+        }, this);
+        return topBar;
```
Comment (frankiefu) [inline app-header/app-header.html]: Consider enhancing the code here so that it exits the loop once the topBar is found.  We can do that later in another PR if you wish.

Comment (blasten) [inline app-header/app-header.html]: :+1:

File: app-header/app-header.html (hunk 3 above)
Comment (blasten) [inline app-header/app-header.html]: you right

File: templates/shrine/src/shrine-app.html (hunk 2 above)
Comment (blasten) [inline templates/shrine/src/shrine-app.html]: yeah, it should. I still need to resize those assets.

File: app-header/app-header.html (hunk 1 above)
Comment (blasten) [inline app-header/app-header.html]: :+1:

File: templates/shrine/src/shrine-app.html (hunk 2 above)
Comment (frankiefu) [inline templates/shrine/src/shrine-app.html]: I am find if you want to do that in a separate PR.

File: templates/shrine/src/shrine-app.html [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+        margin-top: 13px;
+        width: 88px;
+      }
+
+      .spacer {
+        flex: 1 1;
+        -webkit-flex: 1 1;
+        display: flex;
+        align-items: center;
+        justify-content: center;
```
Comment (frankiefu) [inline templates/shrine/src/shrine-app.html]: You don't need `-webkit` prefix for the other stuffs? e.g. `display: -webkit-flex`.... also you might want to just use `@apply(--layout-horizontal);`

Comment (blasten) [inline templates/shrine/src/shrine-app.html]: that's true

File: app-header/app-header.html [hunk 6]
```diff
@@ -245,13 +255,20 @@
        * @type {number}
        */
       _deltaHeight: 0,
+      
+      /**
+       * The offsetTop of `_topBar`
+       *
+       * @type {number}
+       */
+      _topbarTop: 0,
```
Comment (keanulee) [inline app-header/app-header.html]: Change to `_topBarTop`, or rename to `_primaryTop`, `_primaryElTop`, `_primaryElemTop`, or something similar (and similarly to `_topBar`)?

Comment (blasten) [inline app-header/app-header.html]: Done!

File: app-header/app-header.html [hunk 7]
```diff
@@ -245,13 +274,20 @@
...
+      /**
+       * The offsetTop of `_primaryEl`
+       *
+       * @type {number}
+       */
+      _primaryElTop: 0,
 
       /**
-       * The top most app-toolbar.
+       * The element that will remain on top when the header condenses.
```
Comment (keanulee) [inline app-header/app-header.html]: `The element that will remain visible when the header condenses.`

File: app-header/app-header.html [hunk 8]
```diff
@@ -284,6 +320,27 @@
       },
 
       /**
+       * Returns a  a reference to the element that remains visible in condensed mode.
+       *
+       * @return {HTMLElement}?
+       */
+      _getTopBar: function() {
```
Comment (keanulee) [inline app-header/app-header.html]: `_getPrimaryEl`

File: app-header/app-header.html [hunk 9]
```diff
@@ -284,6 +320,27 @@
...
+        var topBar;
```
Comment (keanulee) [inline app-header/app-header.html]: `var primaryEl;`

**Comment by keanulee** [comment]
Just some really minor naming changes. After that LGTM.

**Comment by keanulee** [comment]
LGTM

File: templates/shrine/src/shrine-detail.html [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+
+    <style>
+
+      :host {
+        @apply(--layout-horizontal);
+        -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+      }
+
+      .items {
+        flex: 1 1 35%;
```
Comment (frankiefu) [inline templates/shrine/src/shrine-detail.html]: `-webkit-flex`?

File: templates/shrine/src/shrine-detail.html [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+      .itemsGrid {
+        @apply(--layout-horizontal);
+        @apply(--layout-wrap);
+        @apply(--layout-center-justified);
+        margin-top: 10px;
+      }
+
+      .itemsGrid > .item {
+        flex: 1 calc(50% - 20px);
+        -webkit-flex: 1 calc(50% - 20px);
```
Comment (frankiefu) [inline templates/shrine/src/shrine-detail.html]: prefix should go first

File: templates/shrine/src/shrine-detail.html [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+        margin: 10px;
+        text-decoration: none;
+        background-color: white;
+      }
+
+      .detailSection {
+        @apply(--layout);
+        width: 65%;
+        flex: 1 1 65%;
+        -webkit-flex: 1 1 65%;
```
Comment (frankiefu) [inline templates/shrine/src/shrine-detail.html]: prefix should go first

File: templates/shrine/src/shrine-detail.html [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+        font-weight: 300;
+        line-height: 20px;
+        color: #333;
+      }
+
+      paper-fab {
+        --paper-fab-background: #31f0ef;
+        color: #0a3142;
+        display: inline-flex;
+        display: -webkit-inline-flex;
```
Comment (frankiefu) [inline templates/shrine/src/shrine-detail.html]: prefix should go first

File: templates/shrine/src/shrine-detail.html [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+        }
+
+        .items {
+          @apply(--layout-flex-none);
+          width: 100vw;
+        }
+
+        .itemsGrid > .item {
+          flex: 1 calc(33% - 20px);
+          -webkit-flex: 1 calc(33% - 20px);
```
Comment (frankiefu) [inline templates/shrine/src/shrine-detail.html]: prefix should go first

File: templates/shrine/src/shrine-detail.html [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+       * Phone
+       */
+      @media (max-width: 400px) {
+        .detailSection {
+          @apply(--layout-flex-none);
+        }
+
+        .itemsGrid > .item {
+          flex: 1 calc(50% - 20px);
+          -webkit-flex: 1 calc(50% - 20px);
```
Comment (frankiefu) [inline templates/shrine/src/shrine-detail.html]: prefix should go first

File: templates/shrine/src/shrine-app.html [hunk 16]
```diff
@@ -0,0 +1,375 @@
...
+        --paper-tabs-selection-bar-color: #31f0ef;
+        margin-left: -55px;
+        color: #31f0ef;
+        font-size: 13px;
+      }
+
+      paper-tab {
+        padding: 0;
+        --paper-tab-ink: #31f0ef;
+        flex: none;
```
Comment (frankiefu) [inline templates/shrine/src/shrine-app.html]: -webkit?

File: templates/shrine/src/shrine-list.html [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+      .grid {
+        @apply(--layout-horizontal);
+        @apply(--layout-center-center);
+        @apply(--layout-wrap);
+        margin-top: 10px;
+      }
+
+      .grid > .item {
+        flex: 1 calc(33% - 20px);
+        -webkit-flex: 1 calc(33% - 20px);
```
Comment (frankiefu) [inline templates/shrine/src/shrine-list.html]: prefix should go first

File: templates/shrine/src/shrine-list.html [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+       */
+      @media (max-width: 1200px) {
+        .grid {
+          padding-left: 10px;
+          padding-right: 10px;
+        }
+
+        .grid > .item {
+          flex: 1 calc(50% - 20px);
+          -webkit-flex: 1 calc(50% - 20px);
```
Comment (frankiefu) [inline templates/shrine/src/shrine-list.html]: prefix should go first

**Comment by frankiefu** [comment]
lgtm
